File: Patrones/Comportamiento/Strategy/Python/PagoConBitcoin.py
from EstrategiaDePago import EstrategiaDePago


# Estrategia de pago con Bitcoin
class PagoConBitcoin(EstrategiaDePago):

    def __init__(self, direccion_wallet):
        self.__direccion_wallet = direccion_wallet

    def pagar(self, monto):
        print(f"Pagando {monto} usando Bitcoin. Wallet: {self.__direccion_wallet}")

File: Patrones/Comportamiento/Strategy/Java/src/PagoConBitcoin.java
public class PagoConBitcoin implements EstrategiaDePago {
    private String direccionWallet;

    public PagoConBitcoin(String direccionWallet) {
        this.direccionWallet = direccionWallet;
    }

    @Override
    public void pagar(double monto) {
        System.out.println("Pagando " + monto + " usando Bitcoin. Wallet: " + direccionWallet);
    }
}
